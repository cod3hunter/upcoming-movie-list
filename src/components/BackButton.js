import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';

import Icon from 'react-native-vector-icons/FontAwesome';

const BackButton = props => {
  const { onPress } = props;
  return (
    <TouchableOpacity
      onPress={onPress}
      style={styles.container}>
      <Icon size={18} name='arrow-left' />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'whitesmoke',
    borderRadius: 20,
    elevation: 2,
    zIndex: 10,
    height: 40,
    width: 40,
    position: 'absolute',
    alignItems: 'center',
    justifyContent: 'center',
    left: 16,
    top: 16
  }
});

export default BackButton;