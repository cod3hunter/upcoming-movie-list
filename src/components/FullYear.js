import React from 'react';
import { Text, StyleSheet } from 'react-native';

const FullYear = props => {
  const { dateString } = props;
  return (
    <Text style={styles.text}>
      {(new Date(dateString)).getFullYear() || ''}
    </Text>
  ); 
};

const styles = StyleSheet.create({
  text: { color: 'whitesmoke' }
});

export default FullYear;