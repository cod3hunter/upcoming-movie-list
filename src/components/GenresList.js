import React from 'react';
import { Text, StyleSheet } from 'react-native';

const GenresList = ({ genres }) => {

  return (
    <Text style={styles.text}>
      {
        genres && genres.map((val, index) => index > 0 ? `, ${val.name}` : val.name)
      }
    </Text>
  );
}

const styles = StyleSheet.create({
  text: { color: 'whitesmoke', marginLeft: 10, flex: 1 }
});

export default GenresList;