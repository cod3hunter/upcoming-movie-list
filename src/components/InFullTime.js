import React from 'react';
import { Text, StyleSheet } from 'react-native';

const InFullTime = props => {
  const {minutes} = props;
  if (!minutes) return '';
  const date = new Date(0, 0, 0);
  date.setMinutes(minutes); 

  const inFullTime = `${date.getHours()}h ${date.getMinutes()}min`;
  return (
    <Text style={styles.text}>
      {inFullTime}
    </Text>
  );
};

const styles = StyleSheet.create({
  text: { color: 'whitesmoke', marginLeft: 10 }
});

export default InFullTime;