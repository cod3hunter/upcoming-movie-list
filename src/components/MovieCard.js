import React from 'react';
import { View, Text, StyleSheet, Image, TouchableOpacity } from 'react-native';

import Icon from 'react-native-vector-icons/FontAwesome';

const MovieCard = ({
  title,
  releaseDate,
  imageURI,
  containerStyle,
  voteAvarage,
  voteCount,
  onPress
}) => {
  return (
    <TouchableOpacity
      style={{ ...styles.cardContainer, ...containerStyle }}
      onPress={onPress}
    >
      <Image style={styles.image} source={{ uri: imageURI }} />

      <View style={{ marginLeft: 10, width: '70%' }}>
        <Text style={styles.titleText} numberOfLines={1}>
          {title}
        </Text>
        <Text style={styles.yearText}>
          {new Date(releaseDate).getFullYear()}
        </Text>

        <View style={styles.ratingContainer}>
          <Icon name='star' size={10} color='yellow' />
          <Text style={styles.ratingText}>
            {voteAvarage}
          </Text>

          <Icon name='user' size={10} color='lightgray' style={{ marginLeft: 8 }} />
          <Text style={styles.ratingText}>
            {voteCount}
          </Text>
        </View>

        <View style={styles.releaseDateContainer}>
          <Icon name='calendar' size={15} color='#2089DC' style={{ marginLeft: 8 }} />
          <Text style={styles.releaseDateText}>
            {(new Date(Date.parse(releaseDate)).toUTCString()).replace(' 00:00:00 GMT', '')}
          </Text>
        </View>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  yearText: {
    color: 'lightgray'
  },
  cardContainer: {
    flexDirection: 'row',
    paddingVertical: 10,
  },
  image: {
    width: 80,
    height: 100
  },
  titleText: {
    fontWeight: 'bold',
    fontSize: 15,
    color: 'white',
  },
  ratingContainer: {
    flexDirection: 'row',
    marginTop: 5,
    alignItems: 'center'
  },
  ratingText: {
    color: 'white',
    marginLeft: 5
  },
  releaseDateContainer: {
    alignItems: 'center',
    marginTop: 10,
    flexDirection: 'row'
  },
  releaseDateText: {
    color: '#2089DC',
    marginLeft: 5,
    fontSize: 14
  },
});

export default MovieCard;