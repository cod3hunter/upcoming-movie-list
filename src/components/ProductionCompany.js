import React from 'react';
import { View, Text, Image, StyleSheet } from 'react-native';

const ProductionCompany = ({ imageURI, companyName }) => {
  return (
    <View style={styles.container}>
      <Image
        style={styles.image}
        resizeMode='center'
        source={{ uri: imageURI }}
      />

      <Text style={styles.companyName}>
        {companyName}
      </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    width: '45%',
    backgroundColor: 'whitesmoke',
    justifyContent: 'center',
    alignItems: 'center',
    padding: 16,
    borderRadius: 15,
    marginTop: 8,
  },
  image: {
    height: 100,
    width: 80,
    flex: 1,
  },
  companyName: {
    fontSize: 16,
    marginTop: 5,
    textAlign: 'center'
  }
});

export default ProductionCompany;