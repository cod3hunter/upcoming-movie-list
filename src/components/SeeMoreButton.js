import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';

const SeeMoreButton = props => {
  const {show, onPress} = props;
  return (
    <TouchableOpacity 
      style={{...styles.container, display: show ? 'flex' : 'none'}}
      onPress={onPress}
    >
      <Text style={{ color: 'white' }}>
        See more...
      </Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: { 
    backgroundColor: '#4285F4', 
    padding: 8, 
    alignItems: 'center', 
    marginTop: 10, 
    borderRadius: 5, 
    elevation: 1, 
  }
});

export default SeeMoreButton;