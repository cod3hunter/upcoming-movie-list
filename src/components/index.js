import MovieCard from './MovieCard';
import ProductionCompany from './ProductionCompany';
import BackButton from './BackButton';
import FullYear from './FullYear';
import GenresList from './GenresList';
import InFullTime from './InFullTime';
import SeeMoreButton from './SeeMoreButton';

export { 
  MovieCard, 
  ProductionCompany, 
  BackButton,
  FullYear,
  GenresList,
  InFullTime,
  SeeMoreButton 
};