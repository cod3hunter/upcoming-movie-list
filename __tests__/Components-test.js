import 'react-native';
import React from 'react';

import { BackButton, FullYear, GenresList, InFullTime, MovieCard, ProductionCompany, SeeMoreButton } from '../src/components';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const backButton = renderer.create(<BackButton />).toJSON();
  expect(backButton).toMatchSnapshot();

  const fullYear = renderer.create(<FullYear />).toJSON();
  expect(fullYear).toMatchSnapshot();

  const genresList = renderer.create(<GenresList />).toJSON();
  expect(genresList).toMatchSnapshot();

  const inFullTime = renderer.create(<InFullTime />).toJSON();
  expect(inFullTime).toMatchSnapshot();

  const movieCard = renderer.create(<MovieCard />).toJSON();
  expect(movieCard).toMatchSnapshot();

  const productionCompany = renderer.create(<ProductionCompany />).toJSON();
  expect(productionCompany).toMatchSnapshot();

  const seeMoreButton = renderer.create(<SeeMoreButton />).toJSON();
  expect(seeMoreButton).toMatchSnapshot();
});
