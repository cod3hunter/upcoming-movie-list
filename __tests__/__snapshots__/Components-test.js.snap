// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "whitesmoke",
      "borderRadius": 20,
      "elevation": 2,
      "height": 40,
      "justifyContent": "center",
      "left": 16,
      "opacity": 1,
      "position": "absolute",
      "top": 16,
      "width": 40,
      "zIndex": 10,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 18,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`renders correctly 2`] = `
<Text
  style={
    Object {
      "color": "whitesmoke",
    }
  }
>
  
</Text>
`;

exports[`renders correctly 3`] = `
<Text
  style={
    Object {
      "color": "whitesmoke",
      "flex": 1,
      "marginLeft": 10,
    }
  }
/>
`;

exports[`renders correctly 4`] = `""`;

exports[`renders correctly 5`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "opacity": 1,
      "paddingVertical": 10,
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "height": 100,
        "width": 80,
      }
    }
  />
  <View
    style={
      Object {
        "marginLeft": 10,
        "width": "70%",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "white",
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "lightgray",
        }
      }
    >
      NaN
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "yellow",
              "fontSize": 10,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "marginLeft": 5,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "lightgray",
              "fontSize": 10,
            },
            Object {
              "marginLeft": 8,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "marginLeft": 5,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#2089DC",
              "fontSize": 15,
            },
            Object {
              "marginLeft": 8,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#2089DC",
            "fontSize": 14,
            "marginLeft": 5,
          }
        }
      >
        Invalid Date
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly 6`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "whitesmoke",
      "borderRadius": 15,
      "justifyContent": "center",
      "marginTop": 8,
      "padding": 16,
      "width": "45%",
    }
  }
>
  <Image
    resizeMode="center"
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "flex": 1,
        "height": 100,
        "width": 80,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 16,
        "marginTop": 5,
        "textAlign": "center",
      }
    }
  />
</View>
`;

exports[`renders correctly 7`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#4285F4",
      "borderRadius": 5,
      "display": "none",
      "elevation": 1,
      "marginTop": 10,
      "opacity": 1,
      "padding": 8,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
      }
    }
  >
    See more...
  </Text>
</View>
`;
